src/odemis/acq/stream/_projection.py

```diff
@@ -218,6 +223,437 @@ def _updateImage(self):
+                    pol = None
+                polar_data = self._project2Polar(pos + (pol,))
+
+                # update the histogram
+                # TODO: cache the histogram per image
+                # FIXME: histogram should not include the black pixels outside
+                # of the circle. => use a masked array?
+                # reset the drange to ensure that it doesn't depend on older data
+                self.stream._drange = None
+                self.stream._updateHistogram(polar_data)
+
+                self.image.value = self._project2RGB(polar_data)
+                # TODO why not store the RGB converted image in dict?? Because we need to calc the histogram
+
+        except Exception:
+            logging.exception("Updating %s image", self.__class__.__name__)
+
+    def _onPolarization(self, pos):
+        """
+        Called when the polarization VA has changed.
+        :param pos: (str) Polarization position requested.
+        """
+        self._shouldUpdateImage()
+
+    def _onBackground(self, data):  # TODO why do we need the arg data? Does not work without but dont get why?
+        """
+        Called after the background has changed.
+        :param data: (list) List of data arrays.
+        """
+        # un-cache all the polar images
+        self._polar_dict = {}
+        super(ARRawProjection, self)._onBackground(data)
+
+
+class ARPolarimetryProjection(ARProjection):
+    """
+    An ARProjection is a typical projection used to show 2D angle resolved images.
+    Its .image contains a DataArray 2D RGB (shape YXC), with metadata MD_PIXEL_SIZE and MD_POS.
+    The .background is still on the stream passed in the __init__ as the same background image
+    can be used for multiple projections connected to the same stream.
+    Depending on which type of stream is passed at creation, a more suitable subclass of
+    RGBSpatialProjection might be created (via the use of the __new__ operator).  # TODO still True??
+    That is the recommended way to create a RGBSpatialProjection.
+    """
+
+    def __init__(self, stream):
+        """
+        :param stream: (Stream) The stream the projection is connected to.
+        """
+        super(ARPolarimetryProjection, self).__init__(stream)
+
+        # Cached conversion of the raw 6 polarization images to:
+        #   *Stokes parameters in the detector plane (4 images)
+        #   *Stokes parameters in the sample plane (4 images)
+        #   *E-fields in polar and Cartesian coordinates (5 images)
+        #   *DOPs (degree of polarization) (4 images)
+        # All images in polar representation.
+        self._polarimetry_dict = {}  # dict tuple (float, float, str or None) -> DataArray
+        # Same as above, but the raw (aka rectangular representation) images.
+        self._polarimetry_dict_raw = {}  # dict tuple (float, float, str or None) -> DataArray
+        # TODO  might be useful to have 2 dict one for vis (aka polar rep) the other with raw data
```

Comment:
Isn't it what you've implemented?

Actually this takes a lot of memory. I'm wondering whether it'd be possible to only store one type. Essentially, you want to cache what takes a long time to compute, and not too processed so that if the user changes the display settings, it doesn't have to be recomputed.

Reply:
Yes forgot to remove the TODO. One type does not work as you still would have to do the polar conversion each time, which takes some time...I put a TODO for now to check this again...